crates/sui-json-rpc/src/error.rs

```diff
@@ -94,24 +100,103 @@ impl Error {
                 }
                 _ => RpcError::Call(CallError::Failed(sui_error.into())),
             },
-            Error::QuorumDriverError(err) => match err {
-                QuorumDriverError::NonRecoverableTransactionError { errors } => {
-                    // Note: we probably want a more precise error than `INVALID_PARAMS_CODE`
-                    // but to keep the error code consistent we still use `INVALID_PARAMS_CODE`
-                    let error_object = ErrorObject::owned(
-                        jsonrpsee::types::error::INVALID_PARAMS_CODE,
-                        NON_RECOVERABLE_ERROR_MSG,
-                        Some(errors),
-                    );
-                    RpcError::Call(CallError::Custom(error_object))
-                }
-                _ => RpcError::Call(CallError::Failed(err.into())),
-            },
+            Error::QuorumDriverError(err) => match_quorum_driver_error(err),
             _ => RpcError::Call(CallError::Failed(self.into())),
         }
     }
 }
 
+pub fn match_quorum_driver_error(err: QuorumDriverError) -> RpcError {
```

Comment:
Actually I don't think I can do this due to the orphan rule? Since QuorumDriverError and RpcError are technically from external crates. I recall trying to do this some time earlier and running into this complaint. The workaround is the sui_json_rpc::Error

Reply:
Can we add an implementation of `From<QuorumDriverError>` to the module/crate that `RpcError` is from?

Reply:
RpcError is jsonrpsee::core::Error renamed and not part of the sui repo, so I don't think we can do that. But alternatively I could add an impl. from where QuorumDriverError comes from (sui-types) if this is desired

Reply:
Hmm, it seems wrong to do it that way (QuorumDriver should not be aware of a specific RPC implementation) -- if it's not easy to pull (doesn't make things simpler overall) then let's not re-architect everything to make it work (it's all going to change in the RPC interface redesign anyway).

The way I would have done this is to not have `RpcError` be an alias, rather have it be its own type, that can be converted into `jsonrpsee::core::Error`.